src/main/java/org/reactome/addlinks/fileprocessors/HPAFileProcessor.java

```diff
@@ -0,0 +1,77 @@
+package org.reactome.addlinks.fileprocessors;
+
+import org.reactome.addlinks.EnsemblBiomartUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class HPAFileProcessor extends FileProcessor<List<String>>
+{
+
+    public HPAFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public HPAFileProcessor()
+    {
+        super();
+    }
+
+    private static final int geneNameIndex = 0;
+    private static final int ensemblIdentifierIndex = 2;
+    private static final int uniprotIdentifierIndex = 4;
+
+    /**
+     * Build map of UniProt identifiers to Human Protein Atlas (HPA) identifiers that is used to create HPA cross-references in database.
+     * @return - Map<String, List<String>
+     */
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        String dirToHPAFiles = null;
+
+        try {
+            dirToHPAFiles = this.unzipFile(this.pathToFile);
+        } catch (Exception e) {
+            logger.error("Error unzipping file ({}): {}", this.pathToFile, e.getMessage());
+            e.printStackTrace();
+        }
+        if (dirToHPAFiles != null) {
```

Comment:
With an empty value, this check can be `if (!dirToHPAFiles.isEmpty())`.  It also may be better to use a guard clause here to reduce the nesting:

```
if (dirToHPAFiles.isEmpty()) {
    return new HashMap<>();
}

Map<String, List<String>> mappings = new HashMap<>();
Path inputFilePath = Paths.get(dirToHPAFiles, this.pathToFile.getFileName().toString().replace(".zip", ""));
...

Reply:
Made that change

Reply:
I'm going to disagree with the second point. Having a `return` in the middle of a method can cause confusion. It's common to expect the return near the end of a method rather than near the beginning or middle.

Reply:
That's a fair point.  It is somewhat a matter of style.  I'm not emphatic about using a guard clause in this case, but I do think they are useful and an encouraged refactoring technique:

https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
https://medium.com/better-programming/refactoring-guard-clauses-2ceeaa1a9da
https://refactoring.guru/replace-nested-conditional-with-guard-clauses

Reply:
You could have it like this:

```
    @Override
    public Map<String, List<String>> getIdMappingsFromFile()
    {
        String dirToHPAFiles = getDirToHPAFiles();
        if (dirToHPAFiles.isEmpty()) {
            return new HashMap<>();
        }
        
        Map<String, List<String>> mappings = new HashMap<>();
        
        Path inputFilePath = Paths.get(dirToHPAFiles, this.pathToFile.getFileName().toString().replace(".zip", ""));
        for (String line : getHPAFileLines(inputFilePath)) {
            List<String> tabSplit = Arrays.asList(line.split("\t"));
            String geneName = tabSplit.get(geneNameIndex);
            String ensemblId = tabSplit.get(ensemblIdentifierIndex);
            String uniprotId = tabSplit.get(uniprotIdentifierIndex);
            String hpaUrlId = String.join("-", ensemblId, geneName);

            if (necessaryIdentifiersPresent(geneName, ensemblId, uniprotId)) {
                mappings.computeIfAbsent(uniprotId, k -> new ArrayList<>()).add(hpaUrlId);
            }
        }

        return mappings;
    }
    
    private String getDirToHPAFiles() {
        String dirToHPAFiles = "";

        try {
            dirToHPAFiles = this.unzipFile(this.pathToFile);
        } catch (Exception e) {
            logger.error("Error unzipping file ({}): {}", this.pathToFile, e.getMessage());
            e.printStackTrace();
        }
        
        return dirToHPAFiles;
    }
    
    private List<String> getHPAFileLines(Path inputFilePath) {
        List<String> lines = new ArrayList<>();
        try {
            lines = EnsemblBioMartUtil.getLinesFromFile(inputFilePath, true);
        } catch (IOException e) {
            logger.error("Error reading file ({}): {}", inputFilePath.toString(), e.getMessage());
            e.printStackTrace();
        }
        return lines;
    }
```

or if you would like to keep a single return statement:

```
    @Override
    public Map<String, List<String>> getIdMappingsFromFile()
    {
        Map<String, List<String>> mappings = new HashMap<>();
        
        String dirToHPAFiles = getDirToHPAFiles();
        if (!dirToHPAFiles.isEmpty()) {
            Path inputFilePath = Paths.get(dirToHPAFiles, this.pathToFile.getFileName().toString().replace(".zip", ""));
            for (String line : getHPAFileLines(inputFilePath)) {
                List<String> tabSplit = Arrays.asList(line.split("\t"));
                String geneName = tabSplit.get(geneNameIndex);
                String ensemblId = tabSplit.get(ensemblIdentifierIndex);
                String uniprotId = tabSplit.get(uniprotIdentifierIndex);
                String hpaUrlId = String.join("-", ensemblId, geneName);

                if (necessaryIdentifiersPresent(geneName, ensemblId, uniprotId)) {
                    mappings.computeIfAbsent(uniprotId, k -> new ArrayList<>()).add(hpaUrlId);
                }
            }
        }

        return mappings;
    }
```
Personally, I'm okay with either way in this case.